apple-pie/styles/style.css

```
@@ -13,12 +13,57 @@ body, h1, h2, h3, h4, h5, h6, p, ol, ul {
   margin: 0;
   padding: 0;
   font-weight: lighter;
+
 }
 
-/*  Set the font */
+
+.headerclass {
+  background-image: url(/images/apple-pie.jpg);
```

```suggestion
  background-image: url(../images/apple-pie.jpg);
```
En este caso como la carpeta images esta una ruta anterior, lo que te ha faltado es ponerle los dos puntos para especificarle donde esta correctamente la imagen ! :D